### PR TITLE
chore: update eslint, prettier, eslint-prettier to current

### PR DESCRIPTION
**Pull request by cwilbur**
Tweaked snapshots and eslint configurations slightly to accommodate them.

Docs: Add @cwilbur as a contributor

### CODE REVIEW

File: test/suite/__snapshots__/extension.test.js.snap [hunk 1]
```diff
@@ -3,12 +3,12 @@
 exports[`Extension Test Suite Format document using .prettierrc 1`] = `
 "/* eslint-disable */
 const options = {
-    first: "hello, world",
+    first: 'hello, world',
```
Comment (idahogurl) [inline test/suite/__snapshots__/extension.test.js.snap:6]: This indicates that there is a bug as the snapshots shouldn't change between versions

File: test/suite/__snapshots__/extension.test.js.snap [hunk 2]
```diff
@@ -25,11 +25,11 @@ export { options };
 
 exports[`Extension Test Suite setting "prettierLast" makes prettier run last 1`] = `
 "const options = {
-    first: "hello, world",
+    first: 'hello, world',
```
Comment (idahogurl) [inline test/suite/__snapshots__/extension.test.js.snap:28]: This indicates that there is a bug as the snapshots shouldn't change between versions

File: .envrc [hunk 3]
```diff
@@ -0,0 +1 @@
+layout node
```
Comment (idahogurl) [inline .envrc:1]: What's this file? 🙂

File: package.json [hunk 4]
```diff
@@ -19,7 +19,7 @@
   "version": "5.1.1",
   "engines": {
     "vscode": "^1.63.0",
-    "node": "^16.13.0"
+    "node": "^21.0.0"
```
Comment (idahogurl) [inline package.json:22]: I'd rather not update this as it's not needed by the code or any of the packages.

File: package.json [hunk 5]
```diff
@@ -56,33 +56,34 @@
...
-    "esbuild": "^0.16.14",
-    "esbuild-plugin-text-replace": "^1.2.0",
-    "eslint": "^8.31.0",
+    "@auto-it/exec": "^11.0.4",
+    "@auto-it/first-time-contributor": "^11.0.4",
+    "@auto-it/released": "^11.0.4",
+    "@auto-it/upload-assets": "^11.0.4",
+    "@types/node": "^20.8.6",
+    "@vscode/test-electron": "^2.3.5",
+    "@vscode/vsce": "^2.21.1",
```
Comment (idahogurl) [inline package.json:65]: Why is this package needed? 🙂

File: package.json [hunk 6]
```diff
@@ -56,33 +56,34 @@
...
+    "eslint-plugin-import": "^2.28.1",
+    "eslint-plugin-jest": "^27.4.2",
+    "jest": "^29.7.0",
+    "jest-environment-node": "^29.7.0",
+    "ovsx": "^0.8.3",
+    "prettier": "^3.0.3",
     "ts-node": "^10.9.1",
-    "typescript": "^4.9.5",
+    "typescript": "^5.2.2",
+    "@eslint/eslintrc": "^2.1.2",
```
Comment (idahogurl) [inline package.json:80]: Why is this package needed?

File: package.json [hunk 7]
```diff
@@ -56,33 +56,34 @@
...
-    "prettier": "^2.8.1",
+    "eslint-plugin-import": "^2.28.1",
+    "eslint-plugin-jest": "^27.4.2",
+    "jest": "^29.7.0",
+    "jest-environment-node": "^29.7.0",
+    "ovsx": "^0.8.3",
+    "prettier": "^3.0.3",
     "ts-node": "^10.9.1",
-    "typescript": "^4.9.5",
+    "typescript": "^5.2.2",
```
Comment (idahogurl) [inline package.json:79]: If you update TypeScript then you have to update the examples. Rather we do the least amount of changes

**Comment by DucNg** [comment]
@idahogurl Here is a PR with the minimum changes: https://github.com/idahogurl/vs-code-prettier-eslint/pull/198

**Comment by idahogurl** [comment]
Thank you @cwilbur for your efforts. Unfortunately there are too many changes and just updating the prettier-eslint did not work. It gave an error as I described in this [comment](https://github.com/idahogurl/vs-code-prettier-eslint/issues/171#issuecomment-1783543805)

**Pull request closed without merging**